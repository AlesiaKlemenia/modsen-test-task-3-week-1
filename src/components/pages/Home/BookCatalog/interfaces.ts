export interface ICatalogProps {
  children: JSX.Element[] | JSX.Element;
}
