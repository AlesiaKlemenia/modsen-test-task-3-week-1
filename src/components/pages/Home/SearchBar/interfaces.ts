import { IFilterBarProps } from '@/interfaces/IFilterBarProps';
import { ISearchProps } from '@/interfaces/ISearchProps';

export interface ISearchBarProps extends ISearchProps, IFilterBarProps {}
