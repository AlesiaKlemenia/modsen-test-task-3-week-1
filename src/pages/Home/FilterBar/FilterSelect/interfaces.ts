import { SelectChangeEvent } from '@mui/material';

import { ISelectItem } from '@/interfaces/ISelectItem';

export interface IFilterSelectProps {
  label: string;
  items: ISelectItem[];
  value: string;
  onSelectChange: (e: SelectChangeEvent<unknown>) => void;
}
