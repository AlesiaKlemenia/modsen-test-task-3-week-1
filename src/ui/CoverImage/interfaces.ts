export interface ICoverImageProps {
  title: string;
  coverUrl: string;
}
