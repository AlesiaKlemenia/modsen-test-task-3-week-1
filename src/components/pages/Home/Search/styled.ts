import { Button, TextField } from '@mui/material';
import styled from 'styled-components';

export const SearchFieldWrapper = styled.div`
  width: 35vw;
  min-width: 300px;
  height: 40px;

  display: flex;
  align-items: center;
`;

export const SearchField = styled(TextField)`
  height: 100%;
`;

export const SearchButton = styled(Button)`
  height: 40px;
`;
