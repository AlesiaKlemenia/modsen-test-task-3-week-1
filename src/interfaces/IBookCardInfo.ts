export interface IBookCardInfo {
  id: string;
  title: string;
  bookCategories: string[];
  authors: string[];
  coverUrl: string;
}
