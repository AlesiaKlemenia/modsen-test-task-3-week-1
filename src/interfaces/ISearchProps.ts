export interface ISearchProps {
  onFieldChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
  onKeyDown: React.KeyboardEventHandler<HTMLDivElement>;
  onSearchClick: () => void;
  isError: boolean;
  value: string;
}
