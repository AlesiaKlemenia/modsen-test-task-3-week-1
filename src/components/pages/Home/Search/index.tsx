import { ISearchProps } from '@/interfaces/ISearchProps';

import { SearchButton, SearchField, SearchFieldWrapper } from './styled';

const Search = ({
  onFieldChange,
  onKeyDown,
  onSearchClick,
  isError,
  value,
}: ISearchProps): JSX.Element => (
  <SearchFieldWrapper>
    <SearchField
      placeholder="Enter the book name..."
      fullWidth
      size="small"
      onChange={onFieldChange}
      onKeyDown={onKeyDown}
      error={isError}
      helperText={isError ? 'Can not be empty.' : ''}
    >
      {value}
    </SearchField>
    <SearchButton
      variant="contained"
      sx={{ backgroundColor: 'rgba(0, 0, 0)' }}
      onClick={onSearchClick}
    >
      Search
    </SearchButton>
  </SearchFieldWrapper>
);

export default Search;
