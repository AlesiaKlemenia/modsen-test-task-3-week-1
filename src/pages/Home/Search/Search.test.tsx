import { StyledEngineProvider } from '@mui/material';
import React from 'react';
import renderer, { act } from 'react-test-renderer';

import { ISearchProps } from '@/interfaces/ISearchProps';
import Search from '@/pages/Home/Search';

const mockProps: ISearchProps = {
  onFieldChange: jest.fn(),
  onKeyDown: jest.fn(),
  onSearchClick: jest.fn(),
  isError: false,
  value: '',
};

describe('Search component', () => {
  it('renders correctly with default props', () => {
    const component = renderer.create(
      <StyledEngineProvider injectFirst>
        <Search
          onFieldChange={mockProps.onFieldChange}
          onKeyDown={mockProps.onKeyDown}
          onSearchClick={mockProps.onSearchClick}
          isError={mockProps.isError}
          value={mockProps.value}
        />
      </StyledEngineProvider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calls onFieldChange when input value changes', () => {
    const component = renderer.create(
      <StyledEngineProvider injectFirst>
        <Search
          onFieldChange={mockProps.onFieldChange}
          onKeyDown={mockProps.onKeyDown}
          onSearchClick={mockProps.onSearchClick}
          isError={mockProps.isError}
          value={mockProps.value}
        />
      </StyledEngineProvider>,
    );
    const instance = component.root;

    const searchInput = instance.findByType('input');
    act(() => {
      searchInput.props.onChange({ target: { value: 'Test Book' } });
    });

    expect(mockProps.onFieldChange).toHaveBeenCalledTimes(1);
    expect(mockProps.onFieldChange).toHaveBeenCalledWith({
      target: { value: 'Test Book' },
    });
  });

  it('calls onSearchClick when the search button is clicked', () => {
    const component = renderer.create(
      <StyledEngineProvider injectFirst>
        <Search
          onFieldChange={mockProps.onFieldChange}
          onKeyDown={mockProps.onKeyDown}
          onSearchClick={mockProps.onSearchClick}
          isError={mockProps.isError}
          value={mockProps.value}
        />
      </StyledEngineProvider>,
    );
    const instance = component.root;

    const searchButton = instance.findByType('button');
    searchButton.props.onClick();

    expect(mockProps.onSearchClick).toHaveBeenCalledTimes(1);
  });

  it('displays error message when isError is true', () => {
    const component = renderer.create(
      <StyledEngineProvider injectFirst>
        <Search
          onFieldChange={mockProps.onFieldChange}
          onKeyDown={mockProps.onKeyDown}
          onSearchClick={mockProps.onSearchClick}
          isError
          value={mockProps.value}
        />
      </StyledEngineProvider>,
    );
    const instance = component.root;
    const helperText = instance.findByType('p');
    expect(helperText.children).toContain('Can not be empty.');
  });
});
