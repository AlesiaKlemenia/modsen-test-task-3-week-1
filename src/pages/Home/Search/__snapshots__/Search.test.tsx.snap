// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component renders correctly with default props 1`] = `
<div
  className="sc-beySPh iALRzn"
>
  <div
    className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root sc-guDLey kGuVKR css-wb57ya-MuiFormControl-root-MuiTextField-root"
    onKeyDown={[MockFunction]}
  >
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
        disabled={false}
        id=":r0:"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Enter the book name..."
        required={false}
        type="text"
      />
      <fieldset
        aria-hidden={true}
        className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="css-ihdtdm"
        >
          <span
            className="notranslate"
          >
            â€‹
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium sc-dmyCSP gYqyuN css-sghohy-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[MockFunction]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    Search
  </button>
</div>
`;
